### PR TITLE
replace all uses of Ember.copy, Ember.merge w/ Object.assign

### PR DESCRIPTION
**Pull request by bgentry**
Attempted fix of #834. Not sure if I have to do something weird for the import in [`ember_debug/libs/promise-assembler.js`](https://github.com/emberjs/ember-inspector/pull/835/files#diff-c50ce0bdd9f93deddf67c4549ed160f6) or not…

### CODE REVIEW

**Comment by bgentry** [comment]
FYI, these same test failures appear to also happen on `master`. I'll open a separate issue for that.

File: ember_debug/libs/promise-assembler.js [hunk 1]
```diff
@@ -6,8 +6,9 @@
  */
 
 import Promise from 'ember-debug/models/promise';
+import { copy } from 'ember-copy';
```
Comment (rwwagner90) [inline ember_debug/libs/promise-assembler.js]: @teddyzeenny should this import work here?

Comment (teddyzeenny) [reply ember_debug/libs/promise-assembler.js]: I don't think it would, no. Not unless the app being debugged has `ember-copy` installed as an addon.

Comment (rwwagner90) [reply ember_debug/libs/promise-assembler.js]: So what is your recommendation here?

Comment (teddyzeenny) [reply ember_debug/libs/promise-assembler.js]: There's no way to use Ember addons in `ember-debug` at the moment. We may be able to eventually work this out by custom-building and including the addons in the `ember-debug` final build, but I doubt it's going to be simple. 

For the sake of this PR, my recommendation would just be to not use `Ember.copy` in this file as it's not even doing a deep clone. So either `properties = {...properties}` if our Babel setup supports it, or a simple manual shallow copy by iterating over the properties. Either way, I don't think this function requires an Ember addon.

**Comment by bgentry** [comment]
Would Obect.assign do the trick?

**Comment by teddyzeenny** [comment]
@bgentry yes `Object.assign` works! 👍

**Comment by rwwagner90** [comment]
@teddyzeenny with that in mind, does that work everywhere? Could we completely remove `ember-copy` in favor of that?

**Comment by teddyzeenny** [comment]
`ember-copy` can do deep clones which `Object.assign` and `{ ...spread }` cannot, I think that's the difference? If they're all shallow copies we should be able remove `ember-copy`.

**Comment by bgentry** [comment]
I updated this PR. It no longer uses ember-copy, instead just using `Object.assign` everywhere that `copy` was previously. I'm not certain whether this is ok (if everything is just a shallow copy) so somebody w/ more knowledge should verify that :smile:

**Comment by rwwagner90** [comment]
@bgentry sorry to keep making you change things, but I think we should keep `ember-copy` in the `app` code, but just use `Object.assign` in the `ember_debug` code, just to be safe. With those changes, I would be fine with merging, since @teddyzeenny said `Object.assign` was okay for that one case.

**Comment by rwwagner90** [comment]
As I was typing that, Teddy approved the PR 😂. Let me sync with him offline.

**Comment by rwwagner90** [comment]
@teddyzeenny thinks this is good to go, and we don't need deep copies, so works for me 👍

**Pull request merged**